src/functions.jl

```diff
@@ -146,6 +146,54 @@ function Base.append!(A::NamedDimsArray{L,T,1}, B::AbstractVector) where {L,T}
     return NamedDimsArray{newL}(data)
 end
 
+################################################
+# cat, vcat, hcat
+
+function Base.cat(a::NamedDimsArray{L}, b::AbstractArray; dims) where L
+    numerical_dims = dim(dimnames(a), dims)
+    data = Base.cat(parent(a), b; dims=numerical_dims)
+    newL = unify_names_longest(L, dimnames(data)) # when dims=3 for two 2d arrays
+    return NamedDimsArray{newL}(data)
+end
+
+function Base.cat(a::AbstractArray, b::NamedDimsArray{L}; dims) where L
+    numerical_dims = dim(dimnames(b), dims)
+    data = Base.cat(a, parent(b); dims=numerical_dims)
+    newL = unify_names_longest(L, dimnames(data)) # when dims=3 for two 2d arrays
+    return NamedDimsArray{newL}(data)
+end
+
+function Base.cat(a::NamedDimsArray{La}, b::NamedDimsArray{Lb}; dims) where {La, Lb}
+    newL = unify_names_shortest(La, Lb)
+    numerical_dims = dim(newL, dims)
+    data = Base.cat(parent(a), parent(b); dims=numerical_dims)
+    newL = unify_names_longest(newL, dimnames(data)) # when dims=3 for two 2d arrays
+    return NamedDimsArray{newL}(data)
+end
+
+function Base.cat(a::NamedDimsArray, b::AbstractArray...; dims)
+    return Base.cat(a, Base.cat(b...; dims=dims); dims=dims)
```

Comment:
I really dislike this because it only works if the first argument is NDA. If the first argument is a regular array and any of the other ones is NDA it doesn't work. Any suggestions on a pattern that solves this?

I've asked here: https://stackoverflow.com/questions/63285604/how-to-dispatch-based-on-the-type-of-any-of-the-splattered-args

Reply:
One way `@eval` a fairly long list of possibilities... I guess just `nda, abstractarray`, `a, nda`, `nda,nda` will allow it to be triggered if any of the first two are named. (If you want any-of-first-three, then it's a longer list.) 

Perhaps useful:

https://github.com/mcabbott/AxisKeys.jl/blob/master/src/functions.jl#L93

Reply:
Thanks! It now works if any of the first two args is an NDA.